File: src/frontend/src/services/budgetService.ts
import api from './api';

const budgetService = {
  getBudgets: async (params?: any) => {
    const response = await api.get('/budgets', { params });
    return response.data;
  },

  createBudget: async (data: any) => {
    const response = await api.post('/budgets', data);
    return response.data;
  },

  updateBudget: async (id: number, data: any) => {
    const response = await api.put(`/budgets/${id}`, data);
    return response.data;
  },

  deleteBudget: async (id: number) => {
    const response = await api.delete(`/budgets/${id}`);
    return response.data;
  }
};

export default budgetService;

File: src/backend/src/main/java/com/personal/finance/backend/budgets/service/BudgetService.java
package com.personal.finance.backend.budgets.service;

import com.personal.finance.backend.budgets.dto.request.CreateBudgetRequest;
import com.personal.finance.backend.budgets.dto.request.UpdateBudgetRequest;
import com.personal.finance.backend.budgets.dto.response.BudgetDTO;
import com.personal.finance.backend.budgets.dto.response.BudgetHistoryDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface BudgetService {
    BudgetDTO createBudget(Long userId, CreateBudgetRequest request);
    Page<BudgetDTO> getBudgets(Long userId, Pageable pageable);
    BudgetDTO getBudgetById(Long id, Long userId);
    BudgetDTO updateBudget(Long id, Long userId, UpdateBudgetRequest request);
    void deleteBudget(Long id, Long userId);

    void checkAndAlertBudget(Long userId, Long categoryId, int month, int year);
    List<BudgetHistoryDTO> getBudgetHistory(Long id, Long userId);
}